syntax = "proto3";


service Model {
    rpc getContextLength(Empty) returns (ModelContextLengthResponse) {}
    rpc getDefaultParams(Empty) returns (Params) {}
    rpc getResponseStream(PromptRequest) returns (stream ResponseStream) {}
    rpc getBatchTokenizedLength(LengthRequest) returns (LengthResponse) {}
}

message Empty {}

message Params {
    int32 k = 1;
    int32 top_k = 2;
    float temperature = 3;
    int32 max_new_tokens = 4;
    float score_threshold = 5;
    float repetition_penalty = 6;
    bool use_only_uploaded = 7;
}

message PromptRequest {
    string id = 1;
    string prompt = 2;
    Params params = 3;
}

message ResponseStream {
    string token = 1;
}

message LengthRequest {
    string text = 1;
}

message LengthResponse {
    int32 length = 1;
}

message ModelContextLengthResponse {
    int32 context_length = 1;
}